core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -190,7 +190,7 @@ public Optional<LoadBalancer> selectLoadBalancer(Set<LoadBalancer> loadBalancers
             LOGGER.debug("Found load balancer of type {}", preferredType);
         } else {
             loadBalancerOptional = loadBalancers.stream()
-                .filter(lb -> lb.getType() != null && !preferredType.equals(lb.getType()))
+                .filter(lb -> lb.getType() != null && !LoadBalancerType.OUTBOUND.equals(lb.getType()))
```

Comment:
Just FYI for anyone looking at the code, this line is the only functional change. The rest is refactoring the method name (to try and make it more clear why the outbound load balancer isn't a valid option) and unit tests.